build/scripts/check-toolset-insertion.ps1

```diff
@@ -0,0 +1,126 @@
+Param(
+    [string]$sourcePath = $null,
+    [string]$binariesPath = $null
+)
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+[Void][Reflection.Assembly]::LoadWithPartialName('System.IO.Compression.FileSystem')
```

Comment:
Why do this outside the try / finally?  If it errors wouldn't the script execute successfully?